src/analyzer.ts

```diff
@@ -106,23 +107,20 @@ export class Analyzer {
    * see a mixture of pre-fork and post-fork contents when you analyze with a
    * forked analyzer.
    *
-   * Note: this feature is experimental.
+   * Note: this feature is experimental. It may be removed without being
+   *     considered a breaking change, so check for its existence before calling
+   *     it.
    */
   _fork(options?: ForkOptions): Analyzer {
-    const context =
-        options ? this._context._fork(undefined, options) : this._context;
+    const context = this._context._fork(undefined, options);
     return new Analyzer(context);
   }
```

Comment:
Thought the optional context `_fork` would save an unnecessary cache duplication since it would not be necessary to do until something changed?  It does mean we have to be super sure all non-idempotent analyzer methods cause a context fork, which I think that's how it is intended it to work?  I'll defer to your feelings on this though, since you have a much crisper vision of analyzer's design choices.